crates/oxc_linter/src/rules/jest/consistent_test_it.rs

```diff
@@ -0,0 +1,916 @@
+
+            context(\"suite\", () => { dontTestThis(\"foo\") });
+        ",
+            Some(serde_json::json!([{ "fn": "test", "withinDescribe": "it" }])),
+        ),
+        (
+            "describe(\"suite\", () => { test.skip(\"foo\") })",
+            Some(serde_json::json!([{ "fn": "test", "withinDescribe": "it" }])),
+        ),
+        (
+            "describe(\"suite\", () => { test.concurrent(\"foo\") })",
+            Some(serde_json::json!([{ "fn": "test", "withinDescribe": "it" }])),
+        ),
+        // consistent-test-it with fn=test and withinDescribe=test
+        (
+            "describe(\"suite\", () => { it(\"foo\") })",
+            Some(serde_json::json!([{ "fn": "test", "withinDescribe": "test" }])),
+        ),
+        ("it(\"foo\")", Some(serde_json::json!([{ "fn": "test", "withinDescribe": "test" }]))),
+        // consistent-test-it with fn=it and withinDescribe=it
+        (
+            "describe(\"suite\", () => { test(\"foo\") })",
+            Some(serde_json::json!([{ "fn": "it", "withinDescribe": "it" }])),
+        ),
+        ("test(\"foo\")", Some(serde_json::json!([{ "fn": "it", "withinDescribe": "it" }]))),
+        // consistent-test-it defaults without config object
+        ("describe(\"suite\", () => { test(\"foo\") })", None),
+        // consistent-test-it with withinDescribe=it
+        ("it(\"foo\")", Some(serde_json::json!([{ "withinDescribe": "it" }]))),
+        (
+            "describe(\"suite\", () => { test(\"foo\") })",
+            Some(serde_json::json!([{ "withinDescribe": "it" }])),
+        ),
+        // consistent-test-it with withinDescribe=test
+        ("it(\"foo\")", Some(serde_json::json!([{ "withinDescribe": "test" }]))),
+        (
+            "describe(\"suite\", () => { it(\"foo\") })",
+            Some(serde_json::json!([{ "withinDescribe": "test" }])),
+        ),
+    ];
+
+    let mut fix = vec![
+        // consistent-test-it with fn=test
+        ("it(\"foo\")", "test(\"foo\")"),
+        (
+            "
+                import { it } from '@jest/globals';
+                it(\"foo\")
+            ",
+            "
+                import { it } from '@jest/globals';
+                test(\"foo\")
+            ",
+        ),
+        (
+            "
+                import { it as testThisThing } from '@jest/globals';
+                testThisThing(\"foo\")
+            ",
+            "
+                import { it as testThisThing } from '@jest/globals';
+                test(\"foo\")
+            ",
+        ),
+        ("xit(\"foo\")", "xtest(\"foo\")"),
+        ("fit(\"foo\")", "test.only(\"foo\")"),
+        ("it.skip(\"foo\")", "test.skip(\"foo\")"),
+        ("it.concurrent(\"foo\")", "test.concurrent(\"foo\")"),
+        ("it.only(\"foo\")", "test.only(\"foo\")"),
+        ("it.each([])(\"foo\")", "test.each([])(\"foo\")"),
+        ("it.each``(\"foo\")", "test.each``(\"foo\")"),
+        // Note: couldn't fix
+        // (
```

Comment:
Why this couldn't be fixed?

Reply:
Sorry, I forgot adding the reason, it's the same as the below. Because it couldn't support options. I will add an TODO later.